### ISSUE TITLE
racer doesn't handle importing multiple name from the same module

### ISSUE THREAD

**Issue by AsafMah** [racer-rust/racer#561]
Basically,
typing `use std::cell::`
gives you what you would excpect, Cell, Refcell, etc.

But typing `use std::cell::{` doesn't give you any completions, even though 
`use std::cell::{Cell, RefCell};` is a legitimate line.

As far as I know it produces the same result in all editors, and directly in the command line.

**Comment by yodalee** [comment]
You may try the newest master branch. The leading bracket is handled now #557. The comma (to complete Refcell)  is waiting some testings #559

**Comment by yodalee** [comment]
I dig into this issue for a while, here is some cause I think.
There are three different issues: leading bracket, comma-separated, and space after comma.
In `core.rs` it will call `get_start_of_search_expr`, this will cut down string represents a expression. Since starting bracket is not part of expression, the `use std::cell::{` will become matching a empty string, and then it matches nothing.
I tried to add leading bracket into part of expression ( by adding it to `utils::is_search_expr_char` ), allowing it become `searchstr`. Then in `nameres.rs::search_scope`, I remove bracket from searchstr if it is started with it. This can solve problem one.
I push a branch to my repository, here is the commit diff:
https://github.com/phildawes/racer/commit/9e0825a7fe78e734c5432acd0d00aacdea49c288

I don't like this solution, though. It cannot handle comma-separated and space after comma. If we tried to add comma and space into `utils::is_search_expr_char`, many tests case will be broken. It's not surprised because space shouldn't be part of expression.
Just my feeling. I think more ultimate solution will lies in somewhere in `ast parser` or `nameres::search_scope_headers`. By viewing the running log, the `use` line will trigger a parser in ast by calling `ast::parse_use`. The parser will parse the line with `parse_crate_mod in syntex_syntax`, though it usually failed and return None. There may have some way to let it parse correctly, then the code in `matchers::match_use` can work normally.

I haven't got the whole view of how racer works, and it's just my thought. Maybe someone who knows racer more can give me some hints about how to fix the issues or at least where should I look at.

cc @phildawes @jwilm
